tests/use_case/test_grid_search.py

```diff
@@ -0,0 +1,179 @@
+        grid_search_configurations = self._get_all_grid_search_configurations(grid_search_dictionary)
+
+        grid_search = GridSearch(dataset,
+                                 self.data_preprocessor,
+                                 self.model_trainer,
+                                 grid_search_configurations,
+                                 self.saver)
+
+        # When
+        losses = grid_search.start()
+        configuration_id = list(losses["training_loss"].keys())[0]
+
+        # Then
+        self.assertTrue(losses["training_loss"][configuration_id][grid_search_dictionary["epochs"][0]] > 0.0)
+        self.assertTrue(
+            losses["validation_loss"][configuration_id][grid_search_dictionary["validation_period"][0]] > 0.0)
+        self.assertTrue(losses["test_loss"][configuration_id]["final_epoch"] > 0.0)
+
+        # Tear down
+        self._truncate_table()
+
+    def test_start_for_multiple_batches_of_differing_size(self):
+        # Given
+        dataset_size = 5
+        grid_search_dictionary = {
+            "model": ["RNN"],
+            "epochs": [10],
+            "batch_size": [3],
+            "validation_split": [0.2],
+            "test_split": [0.1],
+            "loss_function": ["MSE"],
+            "optimizer": ["Adagrad"],
+            "time_steps": [1],
+            "validation_period": [5],
+            "scaling_factor": [0.1],
+            "penalty_decimals": [0]
+        }
+        self._insert_test_data(dataset_size)
+        dataset = GraphDataset(self.postgres_connector)
+        grid_search_configurations = self._get_all_grid_search_configurations(grid_search_dictionary)
+
+        grid_search = GridSearch(dataset,
+                                 self.data_preprocessor,
+                                 self.model_trainer,
+                                 grid_search_configurations,
+                                 self.saver)
+
+        # When
+        losses = grid_search.start()
+        configuration_id = list(losses["training_loss"].keys())[0]
+
+        # Then
+        self.assertTrue(losses["training_loss"][configuration_id][grid_search_dictionary["epochs"][0]] > 0.0)
+        self.assertTrue(
+            losses["validation_loss"][configuration_id][grid_search_dictionary["validation_period"][0]] > 0.0)
+        self.assertTrue(losses["test_loss"][configuration_id]["final_epoch"] > 0.0)
+
+        # Tear down
+        self._truncate_table()
+
+    def test_start_a_grid_search(self):
+        # Given
+        dataset_size = 6
+        grid_search_dictionary = {
+            "model": ["RNN"],
+            "epochs": [10, 15],
+            "batch_size": [3, 4],
+            "validation_split": [0.2],
+            "test_split": [0.1],
+            "loss_function": ["MSE"],
+            "optimizer": ["Adagrad"],
+            "time_steps": [1],
+            "validation_period": [5],
+            "scaling_factor": [0.1],
+            "penalty_decimals": [0]
```

Comment:
Fixed values? Dude?

Reply:
Man check the main code, not the tests. The tests are supposed to have fixed values